include/core/Layer.hpp

```diff
@@ -64,6 +80,24 @@ namespace ngen
 
         virtual ~Layer() {}
 
+        /**
+        * @brief Get a class id for this layer object
+        * 
+        * @return int
+        */
+        virtual int class_id() { return LayerClass::kLayer; }
+
+        /**
+         * @brief Get a list of output variables names for this layer
+         * 
+         * @return vector of output variable names
+        */
+
+        virtual std::vector<std::string>&& get_output_variable_names() 
+        { 
+            return std::move(std::vector<std::string>());
+        }
```

Comment:
The call to `std::move` is actually counter-productive: https://devblogs.microsoft.com/oldnewthing/20231124-00/?p=109059